### PR TITLE
fix: account for systems where `musl` is present

### LINKED ISSUE #179: 🐞 bug: `gh-r` for `volta-cli/volta` repo on macOS picks the Windows MSI installer artifact
### Issue description

From the latest Volta release ([v1.0.5](https://github.com/volta-cli/volta/releases/tag/v1.0.5)), it downloads the Windows MSI installer (`volta-1.0.5-windows-x86_64.msi`) instead of the macOS gzipped tarball (`volta-1.0.5-macos.tar.gz`).

```
Downloading volta-cli/volta…
(Requesting `volta-1.0.5-windows-x86_64.msi'…)
[...]
ziextract: WARNING: didn't recognize the archive type of `volta-1.0.5-windows-x86_64.msi'  }(no extraction has been done).%f%b
```

### zinit config

```zsh
zinit ice wait lucid as'program' from'gh-r'
zinit light volta-cli/volta
```

### zinit version or commit ID

de49ebffe89a4b5dad5089e03d1500d802cef7ab

### zsh version

5.8

### host info

CPUTYPE=x86_64 MACHTYPE=x86_64 OSTYPE=darwin21.0 Darwin noct.local 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_X86_64 x86_64

### PR DESCRIPTION
**Pull request by vladdoster**
## Description

- account for systems where `musl` is present
- improve darwin pattern

## Motivation and Context

closes #179 

## Usage examples

N/A

## How Has This Been Tested?

new gh-r tests

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by jankatins** [comment]
6 failures due to GLIBC version:

* bat: https://github.com/sharkdp/bat/issues/2106 -> musl build is static so could be used
* delta: https://github.com/dandavison/delta/issues/1029
* helix: https://github.com/helix-editor/helix/issues/1932 (has musl builds, might work)
* hyperfine: https://github.com/sharkdp/hyperfine/issues/508 (has musl builds, might work)
* mdbook: https://github.com/rust-lang/mdBook/issues/1779
* onefetch: no issue, but the an [even older issue was closed as stale](https://github.com/o2sh/onefetch/issues/175)

**Comment by vladdoster** [comment]
@jankatins I'm aware and have not merged due to it.

**Comment by jankatins** [comment]
@vladdoster  The two additional ubuntu targets are in main, at least my PR fails due to this after i rebased on top of main.

**Comment by vladdoster** [comment]
Yep, it's due to 18.04, but I think it's reasonable to support it since Ubuntu and not EOL support it.

**Comment by jankatins** [comment]
It's more that #227 is now in a non-mergeable state as the CI tests fail :-)

**Comment by jankatins** [comment]
I wonder if it isn't easier in the end to give the user easier control over the bpick pattern, e.g. by adding `bpick_linux` (`bpick_<uname>`) and maybe even `bpick_<uname>_<uname -m>` and then fall back from most specific to least specific to whatever the magic regex is you have now.

**Comment by vladdoster** [comment]
It doesn't require machine learning, lol; it's simple pattern matching with a low number of possible permutations. 

I'm solving an issue I have currently on Ubuntu 22.04 where it doesn't take into account a `linux-gnu` system that has `musl` installed. In 99% of cases, you want the musl binary because it is more portable. This is due to Glibc only being backward compatible with the version it was compiled with. This makes programs like delta crash because my Glibc version is too old, but I can run the musl version with no issues. Also, most systems will have musl installed and should be taken into consideration.

**Comment by vladdoster** [comment]
@jankatins E

> [6 failures ](https://github.com/zdharma-continuum/zinit/runs/6401369888?check_suite_focus=true) due to GLIBC version:
> 
> * bat: [GLIBC 2.29 required by bat_0.20.0_amd64.deb sharkdp/bat#2106](https://github.com/sharkdp/bat/issues/2106) -> musl build is static so could be used
> * delta: [🚀 Continue to support GLIBC_2.28 in binary release dandavison/delta#1029](https://github.com/dandavison/delta/issues/1029)
> * helix: [Feat: support GLIB 2.28 helix-editor/helix#1932](https://github.com/helix-editor/helix/issues/1932) (has musl builds, might work)
> * hyperfine: [Prebuilt binaries require very recent glibc (2.29 released 2019-01) sharkdp/hyperfine#508](https://github.com/sharkdp/hyperfine/issues/508) (has musl builds, might work)
> * mdbook: [Prebuilt `x86_64-unknown-linux-gnu` binaries produce loader errors on systems with older glibc rust-lang/mdBook#1779](https://github.com/rust-lang/mdBook/issues/1779)
> * onefetch: no issue, but the an [even older issue was closed as stale](https://github.com/o2sh/onefetch/issues/175)

This is the exact issue this PR solves. For systems other than rolling release systems, the package manager will probably not have a newer glibc version to satisfy the constraint. However, it can be ran by using the musl version. I'll change it if someone brings up a valid reason, but I have yet to see one.

File: tests/gh-r.zunit [hunk 1]
```diff
@@ -137,27 +129,39 @@
...
 }
-@test 'dive' {
+@test 'dive' { # A tool for exploring each layer in a docker image
   run zinit for @wagoodman/dive; assert $state equals 0
   local dive="$ZBIN/dive"; assert "$dive" is_executable
   $dive --version; assert $state equals 0
 }
-@test 'docker-compose' {
+@test 'docker-buildx' { # A monitor of resources
+  run zinit for as'completions' atclone'buildx* completion zsh > _buildx' lbin'!buildx-* -> buildx' @docker/buildx
```
[the remark this answers is not in the document]
Comment (vladdoster) [reply tests/gh-r.zunit:148]: What do you mean fixed? This worked as it was?

[the remark this answers is not in the document]
Comment (jankatins) [reply tests/gh-r.zunit:148]: No, just that #227 has a commit included which fixes this problem as it showed up there as after the exclamation mark escape fix, the atclone got run and it showed up as a test failure.

**Pull request merged**